lib/screen/onboarding/login_sign_up_screen.dart

```diff
@@ -170,26 +160,25 @@ class _LoginSignUpScreenState extends State<LoginSignUpScreen> {
                       if (_onBoardingStore.isLogin) {
                         final bool canNavigate = await _onBoardingStore.login();
                         if (canNavigate) {
-                          Navigator.of(context).pushNamedAndRemoveUntil(
-                            Routes.dashboard,
-                            (Route<dynamic> route) => false,
-                          );
+                          // add go_router
+                          while (context.canPop()) {
+                            context.pop();
+                          }
+                          GoRouter.of(context).goNamed(RoutesName.dashboard.name);
                         }
                       } else {
-                        final bool canNavigate =
-                            await _onBoardingStore.signUp();
+                        final bool canNavigate = await _onBoardingStore.signUp();
                         if (canNavigate) {
-                          Navigator.of(context).pushNamedAndRemoveUntil(
-                            Routes.dashboard,
-                            (Route<dynamic> route) => false,
-                          );
+                          // add go_router
+                          while (context.canPop()) {
+                            context.pop();
+                          }
+                          GoRouter.of(context).goNamed(RoutesName.dashboard.name);
```

Comment:
```suggestion
                          while (context.canPop()) {
                            context.pop();
                          }
                          context.goNamed(RoutesName.dashboard.name);
```

Reply:
Let's have a similar changes wherever we are adding this code block.

Reply:
To remove repetitive code, maybe we can create an extension method named pushNamedAndRemoveUntil in the utils directory?

Reply:
utils/extension/navigation_extension.dart


```extension GoRouterExtension on GoRouter {
  void goNamedAndRemoveUntil(
    BuildContext context,
    String routeName, {
    Map<String, String>? pathParams,
    Map<String, dynamic>? queryParams,
    Object? extra,
  }) {
    while (context.canPop()) {
      context.pop();
    }
    goNamed(
      routeName,
      pathParameters: pathParams ?? {},
      queryParameters: queryParams ?? {},
      extra: extra,
    );
  }
}

extension GoRouterBuildContextExtensions on BuildContext {
  void pushNamedAndRemoveUntil(
    String routeName, {
    Map<String, String>? pathParams,
    Map<String, dynamic>? queryParams,
    Object? extra,
  }) {
    GoRouter.of(this).goNamedAndRemoveUntil(
      this,
      routeName,
      pathParams: pathParams,
      queryParams: queryParams,
      extra: extra,
    );
  }
}
```

I made extension. but here i need guidance for folder structure.
i write both extenstion in one file.
is it okay or i should need to make another extention file for buildcontext?